packages/shared/src/store/sync.ts

```diff
@@ -268,6 +269,44 @@ export const syncSystemContacts = async (ctx?: SyncCtx) => {
   }
 };
 
+export const syncContactDiscovery = async (ctx?: SyncCtx) => {
+  const systemContacts = await api.getSystemContacts();
+  const phoneNumbers = systemContacts
+    .map((contact) => contact.phoneNumber)
+    .filter((phoneNumber) => phoneNumber && phoneNumber.length > 0) as string[];
+
+  if (!phoneNumbers.length) {
+    logger.trackEvent(AnalyticsEvent.DebugSystemContacts, {
+      context: 'no phone numbers found, skipping discovery',
+    });
+    // this should also mean we no-op on web since we don't have any
+    // system contacts
+    return;
+  }
+
+  try {
+    const matches = (
+      await syncQueue.add('discoverContacts', ctx, () =>
+        api.discoverContacts(phoneNumbers)
+      )
+    ).filter((match) => match[1] !== api.getCurrentUserId());
+    logger.log('got contact discovery matches', matches);
+
+    await db.linkSystemContacts({ matches });
```

Comment:
I know we're updating the local DB contact nickname's to match the system contact, but are we ever propagating that change to the backend? When trying locally, my discovery matches fell back to their on-network nicknames.

Reply:
propagating the nickname change? no, good point

Reply:
fixed in the next commit I'm going to push, but note that we're also going to no-op if we don't have a diff/del set, so you won't see your contact names get updated unless you comment out that early return (or uninstall/reinstall the app).